src/_gettsim/taxes/lohnst.py

```diff
@@ -0,0 +1,401 @@
+from math import trunc
+
+from _gettsim.shared import add_rounding_spec
+from _gettsim.taxes.eink_st import _eink_st_tarif
+
+
+@add_rounding_spec(params_key="lohnst")
+def lohnst_eink(
+    bruttolohn_m: float,
+    steuerklasse: int,
+    eink_st_abzuege_params: dict,
+    vorsorgepauschale: float,
+) -> float:
+    """Calculate tax base for Lohnsteuer (withholding tax on earnings).
+
+    Parameters
+    ----------
+    bruttolohn_m:
+      See basic input variable :ref:`bruttolohn_m <bruttolohn_m>`.
+    steuerklasse:
+      See :func:`steuerklasse`
+    eink_st_abzuege_params:
+      See :func:`eink_st_abzuege_params`
+    vorsorgepauschale
+        See :func:`vorsorgepauschale`
+
+    Returns
+    -------
+
+    """
+    entlastung_freibetrag_alleinerz = (steuerklasse == 2) * eink_st_abzuege_params[
+        "alleinerz_freibetrag"
+    ]
+
+    if steuerklasse == 6:
+        werbungskosten = 0
+    else:
+        werbungskosten = eink_st_abzuege_params["werbungskostenpauschale"]
+
+    if steuerklasse == 6:
+        sonderausgaben = 0
+    else:
+        sonderausgaben = eink_st_abzuege_params["sonderausgabenpauschbetrag"]["single"]
+
+    # Zu versteuerndes Einkommen / tax base for Lohnsteuer.
+    out = max(
+        12 * bruttolohn_m
+        - werbungskosten
+        - sonderausgaben
+        - entlastung_freibetrag_alleinerz
+        - vorsorgepauschale,
+        0.0,
+    )
+
+    return out
+
+
+def _lohnsteuer_klasse5_6_basis(taxable_inc: float, eink_st_params: dict) -> float:
+    """Calculate base for Lst for Steuerklasse 5 and 6.
```

Comment:
Lst = Lohnsteuer?

Would be helpful to describe a bit more what you mean by "base".